ci: slither ignore foundry compilation

fbac opened the pull request:
Ignore foundry compilation when running slither as a CLI and in CI.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Updated GitHub workflows to use `yarn build` instead of `yarn compile` for consistency in build processes.
  - Added configurations to improve workflow execution for Slither and testing.
- **Refactor**
  - Modified `package.json` scripts to streamline testing by removing redundant compile steps.
- **Configuration**
  - Added and adjusted Mocha timeout settings to enhance test reliability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lumtis reviewed (approved):
LGTM

Tests are failing but it doesn't seem to be related to this change

fbac commented:
> @fbac @lumtis nodejs is bumped to 20 and i removed test:prototypes step because it is inlcuded with `yarn test` and i removed `yarn forge` from test workflow for now, when we add more tests we should add it as separate job

LGTM!
